lib/sycamore/sycamore/llms/openai.py

```diff
@@ -251,18 +254,45 @@ def __reduce__(self):
         def deserializer(kwargs):
             return OpenAI(**kwargs)
 
-        kwargs = {"client_wrapper": self.client_wrapper, "model_name": self.model_name}
+        kwargs = {"client_wrapper": self.client_wrapper, "model_name": self._model_name}
 
-        return (deserializer, (kwargs,))
+        return deserializer, (kwargs,)
 
     def is_chat_mode(self):
         return self.model.is_chat
 
+    @staticmethod
+    def _get_cache_key(prompt_kwargs: dict, llm_kwargs: Optional[dict] = None):
+        combined = {"prompt_kwargs": prompt_kwargs, "llm_kwargs": llm_kwargs}
+        json_str = json.dumps(combined, sort_keys=True)
+        hash_obj = hashlib.sha256(json_str.encode("utf-8"))
```

Comment:
This is most likely going to be a common operation! We can use the method from CacheManager.

Reply:
The hashing bit yeah, we can add a common function that accepts a json input to hash. The prompt and llm args are unique to the llm cache

Reply:
true but the cachemanager functions accepts bytes! so the content should not matter.